specification/elastic/resource-manager/Microsoft.Elastic/preview/2022-05-05-preview/elastic.json

```diff
@@ -0,0 +1,2302 @@
+      "post": {
+        "tags": [
+          "TrafficFilter"
+        ],
+        "summary": "Create and Associate private link traffic filter for the given deployment.",
+        "operationId": "createAndAssociatePLFilter_Create",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/MonitorNameParameter"
+          },
+          {
+            "in": "query",
+            "name": "name",
+            "description": "Name of the traffic filter",
+            "type": "string"
+          },
+          {
+            "in": "query",
+            "name": "privateEndpointGuid",
+            "description": "Guid of the private endpoint",
+            "type": "string"
+          },
+          {
+            "in": "query",
+            "name": "privateEndpointName",
+            "description": "Name of the private endpoint",
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "201": {
+            "description": "Success"
+          },
+          "default": {
+            "description": "Default error response.",
+            "schema": {
+              "$ref": "#/definitions/ResourceProviderDefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "createAndAssociatePLFilter_Create": {
+            "$ref": "./examples/PrivateLinkTrafficFilters_Create.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Elastic/monitors/{monitorName}/associateTrafficFilter": {
+      "post": {
+        "tags": [
+          "TrafficFilter"
+        ],
+        "summary": "Associate traffic filter for the given deployment.",
+        "operationId": "AssociateTrafficFilter_Associate",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/MonitorNameParameter"
+          },
+          {
+            "in": "query",
+            "name": "rulesetId",
+            "description": "Ruleset Id of the filter",
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Success"
+          },
```

Comment:
this whole system seems like it should be implemented as resource operations i,e GET\PUT\DELETE of traffic filters and a few operations rather than a bunch of POST operations. Can you please explain why you decided to go with a bunch of POST actions rather than the CRUD operations.


[https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review)

Reply:
We have not modelled this as resource operations as the lifecycle of trafficFilters is not managed by our RP. Instead, it is managed by partner and hence we decided not to have ARM manage the lifecycle. This has already been approved in the PR raised for private swagger repo [link](https://github.com/Azure/azure-rest-api-specs-pr/pull/7741#discussion_r927834075)

Reply:
Ok. Just pointing out that tools on top of ARM like ARM templates \ ARM policy etc do not work well with POST actions. These all depend on the declarative nature of the ARM Commands which does not play well with the scheme here.

Reply:
I understand but this is the expected behavior as of now. We will make a note of this. But these changes are already in production. Is a redesign necessary here ?

Reply:
The PR you are referencing did not introduce these APIs but modified the behaviors slightly. Can u point me to the original PR where all the POST actions were introduced?

Reply:
The comment about modeling the APIs as proper REST APIs was provided in the previous PR as well so the sooner you can fix this the better it will be for your customers.

Reply:
I am approving it for now, but you still have an opportunity to correct this before you go GA. So please consider modelling this properly. If u need assistance schedule a design review with any of the ARM API reviewers